pkg/asset/agent/manifests/agent.go

```diff
@@ -0,0 +1,77 @@
+package manifests
+
+import (
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+const (
+	clusterManifestDir = "cluster-manifests"
+)
+
+var (
+	_ asset.WritableAsset = (*AgentManifests)(nil)
+)
+
+// AgentManifests generates all the required manifests by the agent installer.
+type AgentManifests struct {
+	FileList []*asset.File
+}
+
+// Name returns a human friendly name.
+func (m *AgentManifests) Name() string {
+	return "Agent Manifests"
+}
+
+// Dependencies returns all of the dependencies directly needed the asset.
+func (m *AgentManifests) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&AgentPullSecret{},
+		&InfraEnv{},
+	}
+}
+
+// Generate generates the respective manifest files.
+func (m *AgentManifests) Generate(dependencies asset.Parents) error {
+	for _, a := range []asset.WritableAsset{
+		&AgentPullSecret{},
+		&InfraEnv{},
```

Comment:
I'd not recommend to do that. The current contract of the `Dependencies()` method is to provide just the [types](https://github.com/openshift/installer/blob/6d778f911e79afad8ba2ff4301eda5b5cf4d8e9e/pkg/asset/store/store.go#L199) of the dependencies  - and evenutally allocate a new instance in case of load from disk. So, even though at the cost of a little bit of duplication, the current approach seems to be a reasonable compromise (considering also that you're interested in the concrete asset types, to extract the parsed/loaded data subsequently).